src/Psi.jl

```diff
@@ -0,0 +1,28 @@
+struct BlockWaves{T, Tψ, Basis <: AbstractBasis{T}}
+    basis::Basis
+    # TODO: Find better name, and better way to type it
+    value::Vector{VT} where {VT <: AbstractArray3{Tψ}}
+    n_components::Int
+    n_bands::Int
+end
+function BlockWaves(basis::PlaneWaveBasis, ψ::Vector{VT}; do_nimp=false) where {VT <: AbstractArray3}
+    n_bands = minimum(size.(ψ, 3))
+    # Remove do_nimp, maybe n_bands == nothing if not same number of bands.
```

Comment:
why do you need the field n_bands? Just don't assume the number of bands is constant?